transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/ListTransferDestinationFragment.java

```diff
@@ -0,0 +1,287 @@
+ */
+package com.hyperwallet.android.ui.transfer.view;
+
+import static com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod.TransferMethodFields.TOKEN;
+import static com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod.TransferMethodFields.TRANSFER_METHOD_COUNTRY;
+import static com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod.TransferMethodFields.TYPE;
+import static com.hyperwallet.android.ui.common.view.TransferMethodUtils.getStringFontIcon;
+import static com.hyperwallet.android.ui.common.view.TransferMethodUtils.getStringResourceByName;
+import static com.hyperwallet.android.ui.common.view.TransferMethodUtils.getTransferMethodDetail;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.DialogFragment;
+import androidx.fragment.app.FragmentManager;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProviders;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.hyperwallet.android.model.transfermethod.HyperwalletTransferMethod;
+import com.hyperwallet.android.ui.common.view.OnItemClickListener;
+import com.hyperwallet.android.ui.transfer.R;
+import com.hyperwallet.android.ui.transfer.viewmodel.ListTransferDestinationViewModel;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+public class ListTransferDestinationFragment extends DialogFragment {
+
+    public static final String TAG = "HW:" + ListTransferDestinationFragment.class.getSimpleName();
+    public static final String ARGUMENT_SELECTED_TRANSFER = "SELECTED_TRANSFER";
+
+    private ListTransferDestinationViewModel mListTransferDestinationViewModel;
+    private RecyclerView mRecyclerView;
+    private ListTransferDestinationAdapter mListTransferDestinationAdapter;
+    private View mProgressBar;
+    private HyperwalletTransferMethod mTransferDestination;
+
+    /**
+     * Please don't use this constructor this is reserved for Android Core Framework
+     *
+     * @see #newInstance(HyperwalletTransferMethod)
+     */
+    public ListTransferDestinationFragment() {
+    }
+
+    static ListTransferDestinationFragment newInstance(@NonNull final HyperwalletTransferMethod transferMethod) {
+        Bundle bundle = new Bundle();
+        bundle.putParcelable(ARGUMENT_SELECTED_TRANSFER, transferMethod);
```

Comment:
I think it would be better to set it to the view model, what do you think? Then we wouldn't need to implement onViewStateRestored

Reply:
I thought this is your design but simplified it

Reply:
There's a bit of race condition and observer operation conflict
1) Activity creates the view model, pushed new transfer model value to ViewModel and register observer
   Fragment reference the view model and get value for selected transfer model seems to be null

2) same as 1 except that we register observer so that values are updated and set selected transfer but it will close the activity based on instructions on the observer

3) We simplified it from Transfer model to just the Token so we might not achieve this via Via view model

Reply:
👍